sharding-orchestration/sharding-orchestration-core/src/main/java/io/shardingsphere/orchestration/internal/registry/config/listener/SchemaChangedListener.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2016-2018 shardingsphere.io.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * </p>
+ */
+
+package io.shardingsphere.orchestration.internal.registry.config.listener;
+
+import com.google.common.base.Strings;
+import io.shardingsphere.api.config.rule.RuleConfiguration;
+import io.shardingsphere.orchestration.internal.registry.config.event.DataSourceChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.event.IgnoredChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.event.MasterSlaveRuleChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.event.SchemaAddChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.event.SchemaDeleteChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.event.ShardingRuleChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.node.ConfigurationNode;
+import io.shardingsphere.orchestration.internal.registry.config.service.ConfigurationService;
+import io.shardingsphere.orchestration.internal.registry.listener.PostShardingOrchestrationEventListener;
+import io.shardingsphere.orchestration.internal.registry.listener.ShardingOrchestrationEvent;
+import io.shardingsphere.orchestration.reg.api.RegistryCenter;
+import io.shardingsphere.orchestration.reg.listener.DataChangedEvent;
+import io.shardingsphere.orchestration.reg.listener.DataChangedEvent.ChangedType;
+import io.shardingsphere.orchestration.yaml.ConfigurationYamlConverter;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+/**
+ * Schema changed listener.
+ *
+ * @author panjuan
+ */
+public final class SchemaChangedListener extends PostShardingOrchestrationEventListener {
+    
+    private final ConfigurationService configurationService;
+    
+    private final ConfigurationNode configurationNode;
+    
+    private final Collection<String> existedSchemaNames = new LinkedList<>();
```

Comment:
Why need to record existedSchemaNames here？

Reply:
Because we need to distinguish existed schemas and new one, which will be treated differently.